main.go

```diff
@@ -306,7 +339,31 @@ func protoToSamples(req *prompb.WriteRequest) model.Samples {
 
 func sendSamples(w writer, samples model.Samples) error {
 	begin := time.Now()
-	err := w.Write(samples)
+	shouldWrite := true
+	if election != nil {
+		isLeader, err := election.IsLeader()
+		if err != nil {
+			log.Error("msg", "IsLeader check failed", "err", err)
+			return err
+		}
+		if isLeader {
+			shouldWrite = isLeader
+		} else {
+			isLeader, err := election.BecomeLeader()
+			if err != nil {
+				log.Error("msg", "Error occurred while trying to become a leader", "err", err)
+				return err
+			}
+			shouldWrite = isLeader
```

Comment:
again, seems strange that this is set if both `if` and `else` case instead of outside.